### PR TITLE
added cart management with using auth apis

### PR DESCRIPTION
**Pull request by sashaboi**
Hello all  reviewers. 
completed cart management with user authentication. you can add to cart, increase quantity , decrease quantity , and remove from cart. cart total includes delivery charges. 
Here's the cart screen
screencapture-localhost-3000-cart-2022-03-30-18_17_40
.

### CODE REVIEW

File: src/pages/components/card-card/CartCard.jsx [hunk 1]
```diff
@@ -1,7 +1,91 @@
 import React from 'react'
 import '../card-card/cartcard.css'
+import { useCart } from '../../context/cartcontext'
+import { BsTrashFill } from "react-icons/bs";
+import { useState } from 'react';
```
Comment (Koushith) [inline src/pages/components/card-card/CartCard.jsx:5]: we can move the library imports to the top. just a preference.

File: src/pages/components/product-card/ProductCard.jsx [hunk 2]
```diff
@@ -2,8 +2,62 @@ import React from 'react'
...
+                }
+            } 
+
+
+    const addtocart = (product)=>{
+        setdisabled(true)
+        if (localcart.some((obj)=>obj._id ===product._id)){
+            
+            // eslint-disable-next-line no-useless-concat
+            const urltosend = '/api/user/cart' + '/'+ product._id
```
Comment (Koushith) [inline src/pages/components/product-card/ProductCard.jsx:28]: you can try using ``` `` ``` Template literals.

**Review by Koushith — APPROVED** [review]
Overall code and folder structure looks good. you can remove logs after debugging.

File: src/pages/cart/Cart.jsx [hunk 3]
```diff
@@ -1,36 +1,65 @@
...
-    },
-    (error)=>{
-      console.log(error);
-      
-    })
-  },[]);
+  const delivery = 150
+  const {localcart} = useCart();
+  const pricereducer = (acc,val)=>{
+    acc= acc +(Number(val.price) * Number(val.qty))
```
Comment (ranitasahaa1230) [inline src/pages/cart/Cart.jsx:10]: No need of assigning `acc=`

File: src/pages/context/cartcontext.js [hunk 4]
```diff
@@ -101,63 +102,59 @@ const CartProvider = ({children}) =>{
...
-            
-            })
-            return state;
-
+                // 
+                // 
+                // 
+                
+                
+
```
Comment (ranitasahaa1230) [inline src/pages/context/cartcontext.js]: Remove all these lines

**Review by ranitasahaa1230 — APPROVED** [review]
Suggested some minor changes,
In the deploy preview link, when I'm doing Addtocart, nothing is happening. Do check once
 Rest looks good to me

File: src/pages/cart/Cart.jsx (hunk 3 above)
Comment (sashaboi) [reply src/pages/cart/Cart.jsx:10]: got it , just a return might do. this way of writing sort of helps me read my own code as reducer is a bit tough, will change later on.

File: src/pages/context/cartcontext.js (hunk 4 above)
Comment (sashaboi) [reply src/pages/context/cartcontext.js]: yes.

**Pull request merged**